dataactbroker/handlers/fileHandler.py

```diff
@@ -507,6 +552,9 @@ def checkGeneration(self, submission_id = None, file_type = None):
         """
         if submission_id is None or file_type is None:
             submission_id, file_type = self.getRequestParamsForGenerate()
+        # Check permission to submission
+        self.checkSubmissionById(submission_id, file_type)
```

Comment:
One cent per thought:
1. Decorators are great for signifying something is special about a specific function. I think that if what we're flagging is "this method needs valid permissions", a decorator is good. If instead we're saying "fetch me the permissions", a normal function might be a better bet
2. I'd generally recommend separating permissions checking from business logic. If we performed permissions checks in one place, it's easier to test and maintain. We could then proceed in these functions with that "permissions are good" precondition.